sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -385,23 +385,27 @@ def list_accelerators(
             new_results[acc_name] = acc_info
     results = new_results
 
-    # Unlike other GPUs that can be attached to different sizes of N1 VMs,
-    # A100 GPUs can only be attached to fixed-size A2 VMs,
-    # and L4 GPUs can only be attached to G2 VMs.
-    # Thus, we can show their exact cost including the host VM prices.
-
+    # If we can figure out which instances are to be used then we
+    # can attach them here.
+    accs = list(results.keys())
     acc_infos: List[common.InstanceTypeInfo] = sum(
-        [results.get(a, []) for a in _ACC_INSTANCE_TYPE_DICTS], [])
+        [results.get(a, []) for a in accs], [])
     if not acc_infos:
         return results
 
     new_infos = defaultdict(list)
     for info in acc_infos:
         assert pd.isna(info.instance_type) and pd.isna(info.memory), acc_infos
-        vm_types = _ACC_INSTANCE_TYPE_DICTS[info.accelerator_name][
-            info.accelerator_count]
+        vm_types, _ = get_instance_type_for_accelerator(info.accelerator_name,
+                                                        info.accelerator_count)
+        if vm_types is None:
+            continue
         for vm_type in vm_types:
-            df = _df[_df['InstanceType'] == vm_type]
+            df = _df[(_df['InstanceType'] == vm_type) &
+                     (_df['Region'] == info.region)]
+            if df.empty:
+                logger.warning(f'Cannot find {vm_type} in {info.region}')
+                continue
```

Comment:
Have you encountered any cases when this warning are printed? If so, could you provide it to me?

Reply:
`W 01-06 11:34:37 gcp_catalog.py:403] Cannot find a2-megagpu-16g in asia-northeast1`

I only get this warning when I run `sky show-gpus --cloud gcp --all`. This is why I added this. The accelerator is available in a region but no instance is.

Reply:
This is fixed by adding the `region_filter` as an input to `get_instance_type_for_accelerator`. However that seems a little off maybe there is some inconsistency somewhere in the catalog. Either way it removes the use for this if statement.

Reply:
Thanks for pointing it out! I do find some inconsistency in the catalog and it is tracked in #2953 🫡